packages/data-provider/README.md

```diff
@@ -14,34 +14,111 @@ npm i @overture-stack/lyric
 		audit: {
-			enabled: [INSERT_AUDIT_ENABLED]
-		}
+			enabled: true, // Enable audit functionality (true/false)
+		},
+		recordHierarchy: {
+			pluralizeSchemasName: false, // Enable or disable automatic schema name pluralization (true/false)
+		},
 	},
-	schemaService: {
-		url: [INSERT_LECTERN_URL],
+	idService: {
+		useLocal: true, // Use local ID generation (true/false)
+		customAlphabet: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890', // Custom alphabet for ID generation
+		customSize: 12, // Size of the generated ID
 	},
 	logger: {
-		level: [INSERT_LOG_LEVEL],
+		level: 'info', // Logging level (e.g., 'debug', 'info', 'warn', 'error')
+	},
+	schemaService: {
+		url: 'https://api.lectern-service.com', // URL of the schema service
 	},
 };
 
 
 const lyricProvider = provider(appConfig);
 ```
 
+### Auth Custom Handler
+
+The **authentication custom handler** is a customized function that can be used to verify and manage user authentication within the application. It is used by the auth middleware to process incoming requests.
+
+The handler receives an argument of type `Request` and returns a `UserSessionResult` response type, which provides information about the user's session or any errors encountered during the process.
+
+This result `UserSessionResult` object can include:
+
+- **user**: A `UserSession` object containing details of the authenticated user, which might include fields like `username`, `isAdmin` and `allowedWriteOrganizations`.
+- **errorCode**: A numeric code representing an error that occurred while processing the session request.
+- **errorMessage**: A descriptive message detailing the specific error, if an errorCode is provided.
+
+Example how to implement a custom auth handler:
+
+```javascript
+import { type Request } from 'express';
+import { type UserSessionResult } from '@overture-stack/lyric';
+import jwt from 'jsonwebtoken';
+
+const authHandler = (req: Request): UserSessionResult => {
+    // Extract the token from the request header
+    const authHeader = req.headers['authorization'];
+
+    // Check if the Authorization header exists and starts with "Bearer"
+	if (!authHeader || !authHeader.startsWith('Bearer ')) {
+		return {
+			errorCode: 401,
+			errorMessage: 'Unauthorized: No token provided'
+		}
+	}
+
+	// Extract the token by removing the "Bearer " prefix
+	const token = authHeader.split(' ')[1];
+
+    try {
+		// Verify the token using a public key
+		const publicKey = process.env.JWT_PUBLIC_KEY!;
+		const decodedToken = jwt.verify(token, publicKey);
+
+		return {
+			user: {
+				username: decodedToken.username,
+				isAdmin: decodedToken.isAdmin,
+				allowedWriteOrganizations: decodedToken.scopes,
+			 },
+		};
+	} catch (err) {
+		return {
+			errorCode: 403,
+			errorMessage: 'Forbidden: Invalid token'
+		}
+	}
+};
+```
+
+Add the handler to your `AppConfig` object.
+
+```javascript
+import { AppConfig, provider, UserSession } from '@overture-stack/lyric';
+
+const appConfig: AppConfig = {
```

Comment:
same suggestion as earlier, making this `defaultAuthConfig` would help readability.

also enabled:true by default doesn't feel undebatable. what's the rationale?

Reply:
gah, we're still in the readme. never mind